### PR TITLE
HDDS-7347. [Multi-Tenant] Add proper error message to TenantAssignAdmin and TenantRevokeAdmin

### PR DESCRIPTION
**Pull request by aswinshakil**
## What changes were proposed in this pull request?

- Add error message when the same user is assigned as tenant admin.
- Add error message when the non-admin user is revoked as tenant admin.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7347

## How was this patch tested?

The patch was tested using unit tests.

### CODE REVIEW

**Comment by kerneltime** [comment]
Why can't these APIs be idempotent, if the user is already admin, it should pass, and if the user is not admin, it revoke should pass as well. Why raise an exception?

**Comment by aswinshakil** [comment]
The goal is to propagate back clear messages for these scenarios to improve user experience.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/tenant/OMTenantRevokeAdminRequest.java [hunk 1]
```diff
@@ -116,6 +116,13 @@ public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
           OMException.ResultCodes.INVALID_TENANT_ID);
     }
 
+    // Check if the accessId is already the tenant admin
+    if (!accessIdInfo.getIsAdmin()) {
+      throw new OMException("accessId '" + accessId +
+          "' is already not the tenant '" + tenantId + "' admin.",
```
Comment (smengcl) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/tenant/OMTenantRevokeAdminRequest.java:122]:
```suggestion
          "' is not a tenant admin of '" + tenantId + "'.",
```

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/tenant/OMTenantAssignAdminRequest.java [hunk 2]
```diff
@@ -116,6 +116,13 @@ public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
           OMException.ResultCodes.INVALID_TENANT_ID);
     }
 
+    // Prevent assigning the same user as admin.
+    if (accessIdInfo.getIsAdmin()) {
+      throw new OMException("accessId '" + accessId +
+          "' is already the tenant '" + tenantId + "' admin.",
```
Comment (smengcl) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/tenant/OMTenantAssignAdminRequest.java:122]:
```suggestion
          "' is already a tenant admin of '" + tenantId + "'.",
```

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneTenantShell.java [hunk 3]
```diff
@@ -399,6 +399,14 @@ public void testAssignAdmin() throws IOException {
           + "  \"isDelegatedAdmin\": true\n" + "}\n", true, true);
       checkOutput(err, "", true);
 
+      //Re-assign same admin
```
Comment (smengcl) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneTenantShell.java:402]: nit
```suggestion
      // Assign admin again, expect failure message
```

**Comment by kerneltime** [comment]
> The goal is to propagate back clear messages for these scenarios to improve user experience.

I would still push back against this change. Usually, set ownership or admin access APIs/CLIs are idempotent. The only variation I can imagine is printing that the user was already admin or had access revoked, but the exit/return code/status should be successful.

**Comment by smengcl** [comment]
Thanks @aswinshakil for the effort. Closing this PR without merging it after our discussion with @kerneltime .

**Pull request closed without merging**